src/infra/uuid.rs

```diff
@@ -0,0 +1,33 @@
+use std::str::FromStr;
+
+#[derive(Debug)]
+pub struct Uuid {
+    value: uuid::Uuid
+}
+
+impl TryFrom<String> for Uuid {
+    type Error = uuid::Error;
+
+    fn try_from(value: String) -> Result<Self, Self::Error> {
```

Comment:
Tua função recebe uma `String`, o que faz que o valor seja movido no momento que alguém chamar a função.

Se a pessoa não quiser perder a `String` que ela tem, você vai forçar ela a clonar a `String`, alocando outra `String` igual no heap.

Não tem problema clonar as coisas mas você só precisa de uma `&str` para criar o `Uuid` e não uma `String`, então acho que seria melhor só pedir pra quem chama passar uma `&str`.